File: javascript/binary/binary.js
(function(){

    var Binary = function(binaryString) {
        this._binary = binaryString.match(/^[01]+$/g) ? binaryString : '0';
    };

    Binary.prototype.toDecimal = function() {
        return this._binary.split('').reverse().reduce((old, current, index) => {
            return old + (Math.pow(2,index) * Number(current));
        }, 0);
    };

    module.exports = Binary;

})();

File: java/binary/src/main/java/Binary.java
public class Binary {

	private String input;

	public Binary(String input) {
		this.input = input;
	}

	public int getDecimal() {
		if (!input.matches("[01]+"))
			return 0;
		
		return toDecimal(input, 0);
	}

	private int toDecimal(String ipt, int i) {
		if (i >= ipt.length())
			return 0;
		
		return (int) ((ipt.charAt(ipt.length()-1-i) == '1' ? (Math.pow(2,i)) : 0) + toDecimal(ipt, ++i));
	}

}